source/grid/tria.cc

```diff
@@ -2157,1539 +2158,421 @@ namespace internal
 
 
       /**
-       * Create a triangulation from
-       * given data. This function does
-       * this work for 1-dimensional
-       * triangulations independently
-       * of the actual space dimension.
+       * Create a triangulation from given data.
        */
-      template <int spacedim>
+      template <int dim, int spacedim>
       static void
-      create_triangulation(const std::vector<Point<spacedim>> &v,
-                           const std::vector<CellData<1>> &    cells,
-                           const SubCellData & /*subcelldata*/,
-                           Triangulation<1, spacedim> &triangulation)
+      create_triangulation(const std::vector<Point<spacedim>> &vertices,
+                           const std::vector<CellData<dim>> &  cells,
+                           const SubCellData &                 subcelldata,
+                           Triangulation<dim, spacedim> &      tria)
```

Comment:
This function is really nicely encapsulated now. This is good work by splitting out things into the separate helper data structures.